AzureRmStorageTableCoreHelper.psm1

```diff
@@ -56,6 +56,19 @@ function ExecuteQueryAsync
 			$Results = $Table.ExecuteQuerySegmentedAsync($TableQuery, $token)
 			$token = $Results.Result.ContinuationToken
 			$AllRows += $Results.Result.Results
+			# TakeCount controls the number of results returned per page, not
+			# for the entire query. See e.g. the note in
+			# https://docs.microsoft.com/azure/cosmos-db/table-storage-design-guide#retrieve-large-numbers-of-entities-from-a-query
+			if (($null -ne $token) -and ($null -ne $TableQuery.TakeCount)) {

```

Comment:
This seems good but please let's follow the formatting of most of the code:

```powershell
			if (($null -ne $token) -and ($null -ne $TableQuery.TakeCount))
                        {
				# If the take count is larger than the number of rows in this
				# segment, there are more rows to return.
				if ($TableQuery.TakeCount -gt $Results.Result.Results.Count)
                                {
					$TableQuery.TakeCount -= $Results.Result.Results.Count
				}
                                else
                                {
					# No more rows are available in the current page.
					break
				}
			}
```

Reply:
Fixed.